### PR TITLE
Task/add postgis sink

### PR DESCRIPTION
**Pull request by AlvaroVega**
- [x] PostGIS Sink (including improvements for postgresql backend)
- [x] Doc about new sink https://github.com/telefonicaid/fiware-cygnus/blob/751c59586d6a6da0c323eef2be5711127826a08c/doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_postgis_sink.md
- [x] Tests
To be merged with squash

This introduces a new SInk about PostGIS based in previous sink about PostgreSQL and relies in getGeometry utility:
https://github.com/telefonicaid/fiware-cygnus/blob/master/cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java#L94
which make translations for `geo:type` and `geo:json`.

### CODE REVIEW

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_postgis_sink.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# <a name="top"></a>NGSIPostGISSink
+Content:
+
+* [Functionality](#section1)
+* [Administration guide](#section2)
+* [Programmers guide](#section3)
+
+## <a name="section1"></a>Functionality
+`com.iot.telefonica.cygnus.sinks.NGSIPostGISSink`, or simply `NGSIPostGISSink` is a sink designed to persist NGSI-like context data events within a [PostGIS server](https://postgis.net/) which is relational database exteion of [PostgreSQL server](https://www.postgresql.org/) which allows store GIS objects (Geographic Information Systems). Usually, such a context data is notified by a [Orion Context Broker](https://github.com/telefonicaid/fiware-orion) instance, but could be any other system speaking the <i>NGSI language</i>.
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_postgis_sink.md]: Typo: exteion

File: doc/cygnus-ngsi/flume_extensions_catalogue/README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
     * [NGSIMongoSink](./ngsi_mongo_sink.md)
     * [NGSIMySQLSink](./ngsi_mysql_sink.md)
     * [NGSIPostgreSQLSink](./ngsi_postgresql_sink.md)
+    * [NGSIPostGISSink](./ngsi_postgis_sink.md)
     * [NGSISTHSink](./ngsi_sth_sink.md)
     * [NGSIOrionSink](./ngsi_orion_sink.md)
     * [NGSITestSink](./ngsi_test_sink.md)
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/README.md]: Apart from including the new sink in this file you should also modify the ones shown in this places:

* https://github.com/telefonicaid/fiware-cygnus/pull/1615/files: README.md, cygnus-ngsi/README.md (using 1.12.0 as version), mkdocs.yml
* https://github.com/telefonicaid/fiware-cygnus/pull/1617/files: cygnus-ngsi/README. md again

Comment (AlvaroVega) [reply doc/cygnus-ngsi/flume_extensions_catalogue/README.md]: Boot files are already updated

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_postgis_sink.md (hunk 1 above)
Comment (AlvaroVega) [reply doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_postgis_sink.md]: fixed in 3bd613cdfb7d088d123ffd96fac8e0975eb9d9c7

File: doc/cygnus-ngsi/flume_extensions_catalogue/README.md (hunk 2 above)
Comment (fgalan) [reply doc/cygnus-ngsi/flume_extensions_catalogue/README.md]: imagen

I see modifications in two README.md file, but I think that cygnus-ngsi/README.md (a third one) is pending.

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java [hunk 3]
```diff
@@ -107,8 +107,17 @@ public static String encodeSTHCollection(String in) {
         } // if
         
         if (attrType.equals("geo:json")) {
-            return new ImmutablePair("ST_GeomFromGeoJSON('" + attrValue + "')", true);
+            return new ImmutablePair("ST_GeomFromGeoJSON(" + attrValue + ")", true);
         } // if
+
+        // TBD: What about:  ?
+          // 'geo:line'
```
Comment (manucarrace) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java]: Also geo:point is usually used.

Comment (fgalan) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java]: It seems that geo:point is correctly covered, see https://github.com/telefonicaid/fiware-cygnus/tree/master/cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java#L97

File: doc/cygnus-ngsi/flume_extensions_catalogue/README.md (hunk 2 above)
Comment (AlvaroVega) [reply doc/cygnus-ngsi/flume_extensions_catalogue/README.md]: @fgalan  cygnus-ngsi/README.md was also updated.

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java (hunk 3 above)
Comment (fgalan) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java]: NTC (I guess)

**Pull request merged**